### PR TITLE
Library issues fix

### PR DESCRIPTION
**Pull request by paulovmr**
GUVNOR-2952: [Library] Breadcrumb should be updated when the opened project is renamed
GUVNOR-2947: [Library] Fix create new asset dropdown overflow and order items alphabetically
GUVNOR-2946: [Library] Show business process, guided decision table, data object first at the empty project screen (and order the rest alphabetically).
GUVNOR-2951: [Library] Add option to create new folder/package in a project.

### CODE REVIEW

**Comment by paulovmr** [comment]
Part of an ensemble:

* https://github.com/kiegroup/guvnor/pull/425
* https://github.com/kiegroup/kie-wb-common/pull/739
* https://github.com/kiegroup/drools-wb/pull/428
* https://github.com/kiegroup/jbpm-designer/pull/559

File: guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/AbstractProjectService.java [hunk 1]
```diff
@@ -157,198 +166,204 @@ public WorkingSetSettings loadWorkingSetConfig( final Path project ) {
...
                 throw e;
             } finally {
+                final Project newProject = resourceResolver.resolveProject(Paths.convert(newProjectPath));
+                renameProjectEvent.fire(new RenameProjectEvent(oldProject,
+                                                               newProject));
                 ioService.endBatch();
             }
-            final Project newProject = resourceResolver.resolveProject( Paths.convert( newProjectPath ) );
-            invalidateDMOCache.fire( new InvalidateDMOProjectCacheEvent( sessionInfo, oldProject, oldProjectDir ) );
-            renameProjectEvent.fire( new RenameProjectEvent( oldProject, newProject ) );
```
Comment (paulovmr) [inline guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/AbstractProjectService.java]: Real change is to move this line (and the `newProject` declaration) inside the `finally` block, so it is executed before the `ioService.endBatch();` call. This way, the `renameProjectEvent` is fired before the `deleteProjectEvent`, which is fired when the `pom.xml` is actually removed.

**Comment by manstis** [comment]
@paulovmr Looks OK; however since the sequence of calls is important you could write a Unit Test verifying the order of invocations. See [InOrder](https://static.javadoc.io/org.mockito/mockito-core/2.2.9/org/mockito/InOrder.html).. just a thought.

**Comment by paulovmr** [comment]
@manstis Done, thanks for (all) the reviews!

**Pull request merged**